firebase-firestore/src/main/java/com/google/firebase/firestore/core/Filter.java

```diff
@@ -14,40 +14,12 @@
 
 package com.google.firebase.firestore.core;
 
+import androidx.annotation.NonNull;
 import com.google.firebase.firestore.model.Document;
-import com.google.firebase.firestore.model.FieldPath;
 
-/** Interface used for all query filters. */
 public abstract class Filter {
-  public enum Operator {
-    LESS_THAN("<"),
-    LESS_THAN_OR_EQUAL("<="),
-    EQUAL("=="),
-    NOT_EQUAL("!="),
-    GREATER_THAN(">"),
-    GREATER_THAN_OR_EQUAL(">="),
-    ARRAY_CONTAINS("array_contains"),
-    ARRAY_CONTAINS_ANY("array_contains_any"),
-    IN("in"),
-    NOT_IN("not_in");
-
-    private final String text;
-
-    Operator(String text) {
-      this.text = text;
-    }
-
-    @Override
-    public String toString() {
-      return text;
-    }
-  }
-
-  /** Returns the field the Filter operates over. */
-  public abstract FieldPath getField();
-
   /** Returns true if a document matches the filter. */
-  public abstract boolean matches(Document doc);
+  public abstract boolean matches(@NonNull Document doc);
```

Comment:
We don't use NonNull in the SDK internal classes, only the API layer. We do use Nullable and assume NonNull is the default.